docs/spec/generics.rst

```diff
@@ -333,15 +333,15 @@ argument(s) is substituted.  Otherwise, ``Any`` is assumed.  Example::
 
   class Node(Generic[T]):
```

Comment:
The paragraph above could probably be improved. For instance, without the change I make here, pyright would infer `Node()` to be `Node[None]`, and I think that should be legal.

That said, handling of default values for generics in general is underspecified so maybe we tackle this later (mypy handles them poorly, but in part because it wasn't clear what behaviour should be https://github.com/python/mypy/issues/3737)

Reply:
When this part of the spec was written, mypy defaulted to "implicit-optional", which would give you the same effect as adding `| None`. Since then, the spec was updated to recommend against "implicit-optional", but this code sample wasn't updated. I'd be in favor of just updating the spec to fix the sample.